### PR TITLE
Gear support

### PR DESCRIPTION
**Pull request by tmcgroul**

### CODE REVIEW

File: substrate-processor/src/processor/handlerProcessor.ts [hunk 1]
```diff
@@ -746,6 +868,36 @@ class HandlerRunner<S> extends Runner<S, DataHandlers>{
                         })
                         log.debug('end')
                     }
+                    for (let handler of this.getGearMessageEnqueuedHandlers(handlers, item.event)) {
+                        let event = item.event as GearMessageEnqueuedEvent
+                        let log = blockLog.child({
+                            hook: 'message-enqueued',
```
Comment (eldargab) [inline substrate-processor/src/processor/handlerProcessor.ts:874]: `gear-message-enqueued`

File: substrate-processor/src/processor/handlerProcessor.ts [hunk 2]
```diff
@@ -746,6 +868,36 @@ class HandlerRunner<S> extends Runner<S, DataHandlers>{
...
+                            ...ctx,
+                            log,
+                            event
+                        })
+                        log.debug('end')
+                    }
+                    for (let handler of this.getGearUserMessageSentHandlers(handlers, item.event)) {
+                        let event = item.event as GearUserMessageSentEvent
+                        let log = blockLog.child({
+                            hook: 'message-enqueued',
```
Comment (eldargab) [inline substrate-processor/src/processor/handlerProcessor.ts:889]: `gear-message-sent`

**Review by eldargab — CHANGES_REQUESTED** [review]

File: common/config/rush/pnpm-lock.yaml [hunk 3]
```diff
@@ -682,6 +688,15 @@ packages:
     resolution: {integrity: sha512-k2Ty1JcVojjJFwrg/ThKi2ujJ7XNLYaFGNB/bWT9wGR+oSMJHMa5w+CUq6p/pVrKeNNgA7pCqEcjSnHVoqJQFw==}
     dev: false
 
+  /@gear-js/api/0.24.0_@polkadot+api@7.15.1:
+    resolution: {integrity: sha512-rhHqxhqNFJYPiX36liQV1lPYUzd0tdftH+ldkKRLdkRtBqeEcJjNo0rIMEGRxOSjs9nFtCFZKPRekonahhyYaQ==}
+    peerDependencies:
+      '@polkadot/api': ^8.11.3
```
Comment (dzhelezov) [inline common/config/rush/pnpm-lock.yaml:694]: Here we go again :( 
Is there any way to make this dependency optional only if the gear support is required? `pokadot/api` is really huge and including it as a dependency will make our squid images more heavy + it's know to be very volatile and not very stable.

**Comment by tmcgroul** [comment]
note: the gear team reseted their testnet so the example project is broken:(

**Pull request merged**